monai/networks/trt_wrapper.py

```diff
@@ -0,0 +1,524 @@
+        """
+        e = self.engine
+        ctx = self.context
+
+        last_profile = self.cur_profile
+
+        def try_set_inputs():
+            for binding, t in feed_dict.items():
+                if t is not None:
+                    t = t.contiguous()
+                    shape = t.shape
+                    # TODO: port to new TRT10 API
+                    # mincurmax = list(e.get_profile_shape(self.cur_profile, binding))
+                    # if not self.check_shape(shape, mincurmax):
+                    #    raise ShapeError(f"Input shape to be set is outside the bounds: {binding} -> {shape}")
+                    ctx.set_input_shape(binding, shape)
+                    ctx.set_tensor_address(binding, t.data_ptr())
+
+        while True:
+            try:
+                try_set_inputs()
+                break
+            except ShapeError:
+                next_profile = (self.cur_profile + 1) % e.num_optimization_profiles
+                if next_profile == last_profile:
+                    raise
+                self.cur_profile = next_profile
+                ctx.set_optimization_profile_async(self.cur_profile, stream)
+
+        left = ctx.infer_shapes()
+        assert len(left) == 0
+
+    def infer(self, stream, use_cuda_graph=False):
+        """
+        Runs TRT engine.
+        Note use_cuda_graph requires all inputs to be the same GPU memory between calls.
+        """
+        if use_cuda_graph:
+            if self.cuda_graph_instance is not None:
+                cuassert(cudart.cudaGraphLaunch(self.cuda_graph_instance, stream))
+                cuassert(cudart.cudaStreamSynchronize(stream))
+            else:
+                # do inference before CUDA graph capture
+                noerror = self.context.execute_async_v3(stream)
+                if not noerror:
+                    raise ValueError("ERROR: inference failed.")
+                # capture cuda graph
+                cuassert(
+                    cudart.cudaStreamBeginCapture(stream, cudart.cudaStreamCaptureMode.cudaStreamCaptureModeThreadLocal)
+                )
+                self.context.execute_async_v3(stream)
+                graph = cuassert(cudart.cudaStreamEndCapture(stream))
+                self.cuda_graph_instance = cuassert(cudart.cudaGraphInstantiate(graph, 0))
+                self.logger.info("CUDA Graph captured!")
+        else:
+            noerror = self.context.execute_async_v3(stream)
+            cuassert(cudart.cudaStreamSynchronize(stream))
+            if not noerror:
+                raise ValueError("ERROR: inference failed.")
+
+        return self.tensors
+
+
+class TrtWrappper:
+    """
+    This wrapper implements:
+      - TRT lazy persistent export
+      - Running TRT with optional fallback to Torch
+        (for TRT engines with limited profiles)
+    """
+
+    def __init__(
+        self,
+        model,
+        path,
+        precision="tf32",
```

Comment:
Maybe `fp16` makes more sense for the default?

Thanks.

Reply:
Well initially I was trying to align with TRT defaults. 
From MONAI usage point, fp16 makes better sense, yes. 

Alternative take I thought of was to add trt_args(precision="fp16") function - to be used instead of trt_args dict in configs - found it not much reducing config (separate import needed) and looking less transparent than trt_args dict.